source/common/ssl/context_config_impl.cc

```diff
@@ -170,17 +177,18 @@ Ssl::CertificateValidationContextConfigPtr ContextConfigImpl::getCombinedValidat
 }
 
 void ContextConfigImpl::setSecretUpdateCallback(std::function<void()> callback) {
-  if (tls_certficate_provider_) {
+  if (!tls_certficate_providers_.empty()) {
     if (tc_update_callback_handle_) {
       tc_update_callback_handle_->remove();
     }
     // Once tls_certificate_config_ receives new secret, this callback updates
     // ContextConfigImpl::tls_certificate_config_ with new secret.
-    tc_update_callback_handle_ = tls_certficate_provider_->addUpdateCallback([this, callback]() {
-      tls_certificate_config_ =
-          std::make_unique<Ssl::TlsCertificateConfigImpl>(*tls_certficate_provider_->secret());
-      callback();
-    });
+    tc_update_callback_handle_ =
+        tls_certficate_providers_[0]->addUpdateCallback([this, callback]() {
+          tls_certificate_configs_.clear();
+          tls_certificate_configs_.emplace_back(*tls_certficate_providers_[0]->secret());
+          callback();
+        });
```

Comment:
This will break with multiple certificates, clearing list of configured certificates on each callback, won't it? Though, I guess, this will only happen with SDS, where we support only a single certificate right now.

Maybe add a `TODO` to fix it?

Reply:
Will add TODO.